app/controllers/api/v2/office_controller.rb

```diff
@@ -59,11 +59,14 @@ def search_response(query)
       end
 
       def fetch_and_render_office
-        office = Office.find(params[:id])
-      rescue ActiveRecord::RecordNotFound
-        render status: :not_found, json: not_found_json
-      else
-        render json: office_as_json(office)
+        office = Office.find_by("lower(id) = ?", params[:id].downcase)
+        if office.nil?
+          render status: :not_found, json: not_found_json
+        elsif params[:id] == office.id
+          render json: office_as_json(office)
+        else
+          redirect_to api_v2_office_url(office)
```

Comment:
I'm not close to the code so potentially silly question - now that `redirect_to api_v2_office_url(office)` is added to `fetch_and_render_office`, is `redirect_from_legacy_id_to_new` still going to be used?

Reply:
yes, what that does is if the ID it's passed looks like an old EPi server one then it looks up on the legacy ID and then redirects it, so you still need that for the lookup to look at the other field